ddtrace/internal/telemetry/writer.py

```diff
@@ -227,7 +227,7 @@ def _restart_sequence(self):
     def _stop_service(self, *args, **kwargs):
         # type: (...) -> None
         super(TelemetryBase, self)._stop_service(*args, **kwargs)
-        self.join()
+        self.join(timeout=2)
```

Comment:
I would still skip the join if we know we are stopping the service as part of the restart phase in child processes after a fork.